x/msgfees/keeper/proposal_handler.go

```diff
@@ -1,25 +1,33 @@
 package keeper
 
 import (
+	"fmt"
+
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/provenance-io/provenance/x/msgfees/types"
 )
 
 // HandleAddMsgBasedFeeProposal handles an Add msg based fees governance proposal request
-func HandleAddMsgBasedFeeProposal(ctx sdk.Context, k Keeper, proposal *types.AddMsgBasedFeeProposal) error {
+func HandleAddMsgBasedFeeProposal(ctx sdk.Context, k Keeper, proposal *types.AddMsgBasedFeeProposal, registry codectypes.InterfaceRegistry) error {
 	if err := proposal.ValidateBasic(); err != nil {
 		return err
 	}
 
-	existing, err := k.GetMsgBasedFee(ctx, proposal.Msg.GetTypeUrl())
+	err := checkMsgTypeValid(registry, proposal.MsgTypeUrl)
+	if err != nil {
+		return fmt.Errorf("message type is not a sdk message: %v", proposal.MsgTypeUrl)
+	}
+
```

Comment:
Nice, I had something happening in the cli to check this, but this is where this should actually happen.  We might want to remove it from the cli command then.

Reply:
i saw that and i would like to keep it in the cli too, so that it fails before fees get charged etc..i am thinking good actors here, i don't want them to pay fees